build.gradle

```diff
@@ -109,16 +109,24 @@ subprojects {
     }
 
     // Added for enforcing Java 1.6 version compatibility.
-    // Note that this will not result in compilation failure unless JDK 1.6 or less is used.
-    // GraalVM supports from JDK 1.8; therefore, it doesn't have an effect except IDE inspector
-    // may show incompatible APIs for higher version than JDK 1.6.
     tasks.withType(JavaCompile) {
         def javaVersion = JavaVersion.VERSION_1_6
         sourceCompatibility = javaVersion
         targetCompatibility = javaVersion
-        options.bootstrapClasspath = files("$System.env.JAVA_HOME/jre/lib/rt.jar")
+
+        //TODO: Need to find a better way for setting JAVA6_HOME and 'bootstrapClasspath' automatically.
+        // TODO:If JDK1.6 is not installed in system then should through error because we are enforcing to use JDK1.6
+        // set JAVA6_HOME path to JDK 1.6. [export JAVA6_HOME={jdk1.6 path}
+        if (JavaVersion.VERSION_1_6.equals(javaVersion)
+                && System.env.JAVA6_HOME != null) {
+            options.fork = true
+            options.bootstrapClasspath = files("$System.env.JAVA6_HOME/jre/lib/rt.jar")
+        }
+        else {
+            options.bootstrapClasspath = files("$System.env.JAVA_HOME/jre/lib/rt.jar")
```

Comment:
This just silently uses whatever java version is installed on the system?  Isn't that exactly the opposite of what you're trying to achieve here?

What you want is to ensure that java 1.6 is installed and used to compile, otherwise generate an error (unless the user explicitly asks to use a different version of java).

Reply:
@quinton-hoole 
> This just silently uses whatever java version is installed on the system? Isn't that exactly the opposite of what you're trying to achieve here?

Here when we set java version to **JavaVersion.VERSION_1_6** and set **JAVA6_HOME** to JDK1.6 path then it will take the first case. 
that is --> options.bootstrapClasspath = files("$System.env.**JAVA6_HOME**/jre/lib/rt.jar")
and it will compile code in 1.6 compatible.
Here **JAVA6_HOME** should point to JDK1.6
ex: export JAVA6_HOME=/usr/lib/jvm/JDK1.6


> What you want is to ensure that java 1.6 is installed and used to compile, otherwise generate an error (unless the user explicitly asks to use a different version of java).

When we have only JAVA 1.6 installed in our system and tries to build then gradle is throwing an error by saying that it requires a 1.7+ java version.

So we are using 1.8 JAVA version for Gradle build and we are enforcing the build to use 1.6 java version by setting the below fields explicitly.

<pre>
def javaVersion = JavaVersion.VERSION_1_6
sourceCompatibility = javaVersion
targetCompatibility = javaVersion
        if (JavaVersion.VERSION_1_6.equals(javaVersion)
            &&  (System.env.<b>JAVA6_HOME</b> != null) {
            options.fork = true
            //for 1.6java compatible we have to set JAVA6_HOME to JDK1.6 path
            options.bootstrapClasspath = files("$System.env.<b>JAVA6_HOME</b>/jre/lib/rt.jar") 
        }
        else {
            options.bootstrapClasspath = files("$System.env.JAVA_HOME/jre/lib/rt.jar")
        }
</pre>
Currently, with **JAVA6_HOME**  setting to JDK1.6 path, we can catch the code which does not comply with java1.6 version.

Here I am working for how we can set **options.bootstrapClasspath** to JDK1.6/jre/lib/rt.jar  path in the JDK1.6 case without system env variable(JAVA6_HOME)

Reply:
Please ask @VenuReddy2103 to explain.  
My point was that the desired java version to compile Amino with should be explicitly specified somewhere in the gradle config files that users check out of git.  And if it's not specified, and error should be generated.

Right now, if someone deletes the statement:

`def javaVersion = JavaVersion.VERSION_1_6`

or if they do not have JAVA6_HOME set in their environment, it is not well defined which java version will be used to compile with.  I different version could be used on each developers' machine, which is what we're trying to avoid.